rmw_opensplice_cpp/src/qos.cpp

```diff
@@ -56,6 +58,35 @@ bool set_entity_qos_from_profile_generic(
       return false;
   }
 
+  if (qos_profile.depth != RMW_QOS_POLICY_DEPTH_SYSTEM_DEFAULT) {
+    if (qos_profile.depth > static_cast<size_t>((std::numeric_limits<DDS::Long>::max)())) {
+      RMW_SET_ERROR_MSG(
+        "failed to set history depth since the requested queue size exceeds the DDS type");
+      return false;
+    }
+    // ensure the history depth is at least the requested queue size
+    if (kind == DDS::KEEP_LAST_HISTORY_QOS &&
+        static_cast<size_t>(depth) < qos_profile.depth)
+    {
+      depth = static_cast<DDS::Long>(qos_profile.depth);
+    }
+  }
+
+  return true;
+}
+
+template<typename DDSEntityQos>
+bool set_entity_qos_from_profile_generic(
+  const rmw_qos_profile_t & qos_profile,
+  DDSEntityQos & entity_qos)
+{
+  if (!rmw_qos_to_history(
+                qos_profile,
+                entity_qos.history.kind,
+                entity_qos.history.depth)) {
```

Comment:
nitpick: this looks like an arbitrary amount of indention

I'm not sure if this package is running our style linter or not (CI will complain if it is), but I'd expect something more like:

```c++
  if (!rmw_qos_to_history(
    qos_profile,
    entity_qos.history.kind,
    entity_qos.history.depth))
  {
    return false;
  }
```

Again, a nitpick and only needs to be addressed if the linter is turned on and it complains.